lib/zendesk_apps_tools/deploy.rb

```diff
@@ -56,7 +55,7 @@ def upload(path)
         uploaded_data: Faraday::UploadIO.new(package_path, 'application/zip')
       }
 
-      response = connection(:multipart).post('/api/v2/apps/uploads.json', payload)
+      response = get_connection(:multipart).post('/api/v2/apps/uploads.json', payload)
```

Comment:
I think it may be unclear when reading this why we sometimes use `connection` and other times `get_connection`. What if we kept a cache keyed by the encoding?
e.g.
```
def cached_connection(encoding = :url_encoded)
  @connection ||= {}
  @connection[encoding] ||= get_connection(encoding)
end
```

Reply:
wow this is pretty cool. 

1) Say we get a `cached_connection(:url_encoded)` the first time,
* the first line becomes `@connection = {}`
* the second line becomes `@connection[:url_encoded] = get_connection(:url_encoding)`

2) The next time someone calls cache_connection(:url_encoded),
* the first line evaluates to just `@connection` because it is already defined
* the second line returns `@connection[:url_encoded]` which was assigned previously

3) The next next time someone calls cache_connection(:multipart),
* the first line evaluates to just`@connection` again because it is already defined
* however the second line becomes `@connection[:multipart] = get_connection(:multipart)`

Now it has  both values cached - Brilliant! Thanks, @danielbreves

Reply:
<img width="1280" alt="screen shot 2018-12-13 at 10 25 43 pm" src="https://user-images.githubusercontent.com/17760485/49936049-f2136e80-ff26-11e8-904e-15726936fd5c.png">